addons/account/models/account_move.py

```diff
@@ -1994,6 +1994,30 @@ def action_invoice_register_payment(self):
             .with_context(active_ids=self.ids, active_model='account.move', active_id=self.id)\
             .action_register_payment()
 
+    def action_switch_invoice_into_refund_credit_note(self):
+        for move in self:
+            if move.type not in ['in_invoice', 'out_invoice']:
+                raise ValidationError(_("This action isn't available for this document."))
+            
+        for move in self.filtered(lambda move: move.type in ['in_invoice', 'out_invoice']):
+            move.type = move.type.replace('invoice', 'refund')       
+            reversed_move = move._reverse_move_vals({}, False)
+            new_line_ids = []
+            for line_ids in reversed_move['line_ids']:
+                line_tmp = line_ids[2]
+                line_tmp['amount_currency'] = abs(line_tmp['amount_currency'])
+                line_tmp['quantity'] = abs(line_tmp['quantity'])
+                line_tmp['price_unit'] = abs(line_tmp['price_unit'])
+                line_tmp['discount'] = abs(line_tmp['discount'])
+                if move.amount_total < 0:
+                    new_credit = line_tmp['debit']
+                    new_debit = line_tmp['credit']
+                    line_tmp['credit'] = new_credit
+                    line_tmp['debit'] = new_debit
+                new_line_ids.append((0, 0, line_tmp))
+            move.write({'line_ids': [(5, 0, 0)]})
+            move.write({'line_ids': new_line_ids})
```

Comment:
In my opinion, it should be something like:
```
for move in self:
    reversed_move = move._reverse_move_vals({}, False)
    if move.amount_total < 0:
        # Inverse all invoice_line_ids.
        for _, _, line_vals in reversed_move['invoice_line_ids']:
            line_vals.update({
                'quantity': -line_vals['quantity'],
                'amount_currency': -line_vals['amount_currency'],
                'debit': line_vals['credit'],
                'credit': line_vals['debit']
            })
    move.write({'invoice_line_ids': [(5, 0)] + reversed_move['invoice_line_ids']})
```
Because using `abs` lead to some bugs in this situation:
 - If all business fields are negatives => the `amount_total` becomes different, it's confusing for the user.
 - If the `amount_currency` is negative but the `amount_total` is positive => bug as `amount_currency` & `balance` not longer share the same sign.

Note: I think it's a better idea to inverse only the `invoice_line_ids` and let the `onchange` to recompute the taxes / cash rounding / payment terms.